dev/image-guidelines.md

```diff
@@ -0,0 +1,329 @@
+# Guidelines for creating image content
+
+This document provides guidance for creating image content for the Velociraptor documentation.
+It specifies some informal standards and gives some tool-specific advice with the goal of ensuring as much consistency as possible in our image content.
+
+---
+
+* [Image categories](#image-categories)
+* [Fonts](#fonts)
+* [Colors](#colors)
+* [Excalidraw object library](#excalidraw-object-library)
+* [Screenshots](#screenshots)
+   * [Screen size](#screen-size)
+      * [Partial screens](#partial-screens)
+      * [Full screens](#full-screens)
+   * [Velociraptor theme](#velociraptor-theme)
+      * [Theme tweaks](#theme-tweaks)
+   * [Annotations](#annotations)
+      * [Arrows](#arrows)
+      * [Numbered callouts](#numbered-callouts)
+      * [Excalidraw-specific options](#excalidraw-specific-options)
+* [Line diagrams](#line-diagrams)
+* [Exporting images](#exporting-images)
+   * [Formats and settings](#formats-and-settings)
+   * [Fonts in SVG images](#fonts-in-svg-images)
+   * [File size](#file-size)
+
+---
+
+## Image categories
+
+Image content generally falls into one of these categories:
+
+1. [Screenshots](#screenshots)
+	- unannotated
+	- annotated
+2. [Line diagrams](#line-diagrams) (flow charts, sequence diagrams, etc.)
+
+For **annotated screenshots** and **line diagrams** we use Excalidraw because it's easy to use and it produces drawings with a realistic hand-drawn effect.
+For annotated screenshots this aspect contrasts well with inorganic GUI elements, and in general it's humanistic and visually appealing.
+
+For **unannotated screenshots** we use the PNG format since it has lossless compression which preserves the legibility in text in images of the Velociraptor GUI. There's no need to do anything fancy with these images. Just try to keep the size below 300KB if possible.
+
+## Fonts
+
+The standard Excalidraw version uses a default font named Virgil which is too scrawly and therefore a bit hard to read (as discussed [here](https://github.com/excalidraw/excalidraw/issues/2945)).
+The default fonts are not able to be changed without editing the application's source, and the Excalidraw developers have stated disinterest in make custom fonts possible.
+
+Fortunately there exists a reasonably well-maintained fork which allows adding custom fonts
```

Comment:
Im a bit reluctant to use an unofficial fork - do we absolutely need it? Dont we replace the fonts anyway?

The convenience with the official project is that there is a free site we can use without needing to mess with yarn - this makes it a lot more accessible.

Reply:
The official version doesn't allow custom fonts at all. That means you can't see an accurate representation of what you're working on while you're working on it.

All the fonts are also variable width so there will definitely be significant differences if we swap a font with a _different_ font, especially given that the official version's default "Virgil" is a very wide font. When we set the font URL in the SVG we're actually undoing the swap-to-defaults that the SVG export performs.

Looking at the SVGs that you recently created, which I assume came from the official hosted app, they have one font referenced:
![Screenshot from 2024-09-18 16-39-56](https://github.com/user-attachments/assets/0ce87cf2-81c0-402a-852f-ede5272b3ed6)
There's no indication that the embedded font there is actually "Virgil".

I've just done a quick test now at excalidraw.com: added text in one of each of their 3 fonts, then exported to SVG. The file contains an embedded "Excalifont", _5 embedded copies_ of "Nunito", and "Comic Shanns" (which is supposed to be their code font but looks not at all code-like). Until now their fonts have always been Virgil (now rebranded to Excalifont), Cascadia, and Assistant.

So problem is really the opposite, where using the hosted app means they can change things on a whim and and offer no guarantees of backward compatibility. The fork doesn't have any issues - all the functionality works - and can always pin a version if necessary. I don't think there will be any new features in the official version that we will be yearning for - the app does what's needed.

Reply:
I understand the layout can differ but is it really that important?

I generally prefer things that take least effort for the person developing the doc and leave more of the look and feel to the site builder scripts - that way there can be less variability and the user only needs to think about what to write not how to lay the text out.  In other words it does not matter if the text is slightly bigger or smaller as long as the bounding box is set over the correct region.

Reply:
If the bounding box is configured with extra room for text expansion that partly mitigates the problem. That is IF the user constantly remembers to do this for every bounding box.

Even so there are situations where it doesn't work:

1. If text is left-aligned but manually positioned to be centered under another object, then the font changes size and the text extended to the right and possibly even wraps. It leaves an unbalanced result that wasn't the author's intention.
![Screenshot from 2024-09-19 12-27-14](https://github.com/user-attachments/assets/b4eba545-7436-4ece-895a-ebad28a4163a)

2. If the text is associated with an object on it's right and the text is not set to right-align, then swapping can cause it to overlap the object on the right or shrink away for the object leaving a big gap. So that means every time you have text associated with something on it's right you have to remember to make the text right aligned and extend the bounding box to the left.
![Screenshot from 2024-09-19 12-32-28](https://github.com/user-attachments/assets/4d205cda-b1a3-4e5e-b978-2dafa53e5b8e)

Keeping all these possibilities in mind sounds like much more effort to me. I never have to worry about whether the bounding boxes will be sufficient or if I've left some text left-aligned when it properly should be right-aligned. WYSIWYG is really really important.